pkg/apis/core/validation/validation.go

```diff
@@ -5279,3 +5280,32 @@ func ValidateProcMountType(fldPath *field.Path, procMountType core.ProcMountType
 		return field.NotSupported(fldPath, procMountType, []string{string(core.DefaultProcMount), string(core.UnmaskedProcMount)})
 	}
 }
+
+// validateTopologySpreadConstraints validates given TopologySpreadConstraints.
+func validateTopologySpreadConstraints(constraints []core.TopologySpreadConstraint, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	supportedRespModes := sets.NewString(string(core.DoNotSchedule), string(core.ScheduleAnyway))
+	// tuple {topologyKey, whenUnsatisfiable} denotes one kind of constraint
+	allConstraintKinds := sets.String{}
+	for i, constraint := range constraints {
+		subFldPath := fldPath.Index(i)
+		if val := constraint.MaxSkew; val <= 0 {
```

Comment:
why do we need to check this , lets make this unsigned

Reply:
with `uint32`, it also needs to be validated to be non-zero.

Reply:
fyi @krmayankk: Please do not use uint types for restricting the domain to positive integers--it encourages math errors. The code is correct.

Reply:
@lavalamp  could you elaborate ? underflow/overflow can occur in both . Keeping it unsigned means i dont need to validate if its negative 

@Huang-Wei why should it be validated to be non zero ? It can be zero, which means perfect spreading with no skew

Reply:
at zero, it will be impossible to legally add another pod. So zero can't be allowed.

underflow can occur always, but underflow at 0 is vastly more likely to be hit in normal code. e.g. `if x - y > z` doesn't do what people expect any more.

Reply:
We can't use zero to describe a "perfect" balanced case - think about how you can spread 2 pods into 2 zones if it's set to zero.

It's similar to the definition of "perfect balanced tree" - where we usually define it as "height of the left and right subtrees from every node differ by 1(maxSkew) or less"

Reply:
thanks @lavalamp @Huang-Wei for the explanation and it makes sense.  Currently the scheduler decides on a per pod basis .For a hypothetical use case, where an application cannot tolerate unbalance by even one pod, what would be ideal is to allow the scheduler to schedule in batches. So in your example of two pods and two zones, the next scheduling can happen for 2 or 4 more pods otherwise it fails. May be we support this when we have batch scheduling (i think there was some work going on). 

We can even argue, whether we should even design such applications which cannot tolerate an imbalance of 1 pod .